### DISCUSSION TITLE
Get list of unloadable sensors/schedules via GraphQL or DagsterInstance

### DISCUSSION THREAD

**Discussion by CHB-BH** [dagster-io/dagster#19129] [Q&A]
(UPDATE) We figured out how to fetch the data via GraphQL. Still not sure about DagsterInstance.

Hi Dagster Team,

Is it possible to get a list of sensor/schedule that are still ACTIVE, but the codebase/definition is missing (i.e. the code is unloadable)?

Ideally, via GraphQL or DagsterInstance (e.g. OpExecutionContext.instance)

e.g. if we have an existing running sensor/schedule, but we deleted the sensor/schedule definition and deploy a new code version, then the Schedules/Sensor page will show an alert (see example below)

Thank you.

![image](https://github.com/dagster-io/dagster/assets/152458912/e0fa9534-b52a-48ba-b22a-3f0195b25d3b)

(Update) After searching through the GraphQL documentation, I can find `unloadableInstigationStatesOrError`

```graphql
query SchedulesQuery {
  unloadableInstigationStatesOrError(instigationType: SCHEDULE){
    ... on InstigationStates{
      results {
        id
        name
      }
    }
  }
}

```

Is there a `DagsterInstance` equivalent of the above GraphQL? e.g.

```python
@op
def dagster_op(context: OpExecutionContext):
    context.instance.unloadableInstigationStatesOrError()   # is it possible?
```

**Comment by OwenKephart** [comment]
@CHB-BH In this case, it might be easiest to just invoke that gql query from the body of the op.

The implementation of the resolver for isn't particularly simple: https://sourcegraph.com/github.com/dagster-io/dagster/-/blob/python_modules/dagster-graphql/dagster_graphql/schema/roots/query.py?L688:17-688:59.
